src/core/mac/mac_frame.cpp

```diff
@@ -1060,7 +1060,8 @@ uint16_t Frame::GetFcsSize(void) const
 
 // LCOV_EXCL_START
 
-#if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_INFO) && (OPENTHREAD_CONFIG_LOG_MAC == 1)
+#if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_INFO || OPENTHREAD_CONFIG_ENABLE_DYNAMIC_LOG_LEVEL) \
```

Comment:
I don't think we need this. `DYNAMIC_LOG_LEVEL` only lets us reduce the log level from the pre-set `CONFIG_LOG_LEVEL`  (e.g., if we compile with `LOG_LEVEL=INFO` and have `DYNAMIC_LOG_LEVEL` we can reduce it to `NOTE` but cannot change it to `DEBG`)
```
/**
 * @def OPENTHREAD_CONFIG_ENABLE_DYNAMIC_LOG_LEVEL
 *
 * Define as 1 to enable dynamic log level control.
 *
 * Note that the OPENTHREAD_CONFIG_LOG_LEVEL determines the log level at
 * compile time. The dynamic log level control (if enabled) only allows
 * decreasing the log level from the compile time value.
 *
 */
#ifndef OPENTHREAD_CONFIG_ENABLE_DYNAMIC_LOG_LEVEL
#define OPENTHREAD_CONFIG_ENABLE_DYNAMIC_LOG_LEVEL 0
#endif
````

Reply:
This was recently changed to `NOTE` level with #3962.